src/ert/storage/local_ensemble.py

```diff
@@ -574,13 +662,122 @@ def load_parameters(
             Loaded xarray Dataset with parameters.
         """
 
-        return self._load_dataset(group, realizations)
+        # UGLY AF, would be better to narrow in realizations
+        # to be either None of a tuple of ints
+        drop_reals_dim = False
+        selected_realizations: Union[None, int, List[int]] = []
+        if realizations is None:
+            selected_realizations = None
+        elif type(realizations) is int:
+            assert type(realizations) is int
+            drop_reals_dim = True
+            selected_realizations = realizations
+        elif type(realizations) is np.ndarray:
+            selected_realizations = realizations.tolist()
+        elif isinstance(realizations, tuple):
+            selected_realizations = list(realizations)
+        elif isinstance(realizations, list):
+            selected_realizations = realizations
+        else:
+            raise ValueError(f"Invalid type for realizations: {type(realizations)}")
+
+        try:
+            ds = self.open_unified_parameter_dataset(group)
+            if selected_realizations is None:
+                return ds
+
+            return ds.sel(realizations=selected_realizations, drop=drop_reals_dim)
+
+        except (ValueError, KeyError, FileNotFoundError):
+            # Fallback to check for real folder
+            try:
+                if selected_realizations is None:
+                    return xr.combine_nested(
+                        [
+                            xr.open_dataset(p)
+                            for p in self._path.glob(f"realization-*/{group}.nc")
+                        ],
+                        concat_dim="realizations",
+                    )
+                elif type(selected_realizations) is int:
+                    return xr.open_dataset(
+                        self._path
+                        / f"realization-{selected_realizations}"
+                        / f"{group}.nc"
+                    )
+                else:
+                    assert type(selected_realizations) is list
+                    return xr.combine_nested(
+                        [
+                            xr.open_dataset(
+                                self._path / f"realization-{real}" / f"{group}.nc"
+                            )
+                            for real in selected_realizations
+                        ],
+                        concat_dim="realizations",
+                    )
+            except FileNotFoundError as e:
+                raise KeyError(
+                    f"No dataset '{group}' in storage for "
+                    f"realization {selected_realizations}"
+                ) from e
+
+    def _find_unified_dataset_for_response(self, key: str) -> str:
+        all_gen_data_keys = {
+            k
+            for k, c in self.experiment.response_configuration.items()
+            if type(c) is GenDataConfig
+        }
+
+        if key == "gen_data" or key in all_gen_data_keys:
```

Comment:
Should use the names from the config objects?

Reply:
Could, but it might involve a bit more and be suitable for a different PR?